pkg/cluster/backup_manager.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/coreos/etcd-operator/pkg/cluster/backupstorage"
+	"github.com/coreos/etcd-operator/pkg/spec"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+)
+
+type backupManager struct {
+	logger *logrus.Entry
+
+	clusterConfig Config
+	backupPolicy  *spec.BackupPolicy
+	restorePolicy *spec.RestorePolicy
+	s             backupstorage.Storage
+}
+
+func (bm *backupManager) setup() error {
+	var err error
+	bm.s, err = bm.createStorage()
+	if err != nil {
+		return err
+	}
+
+	r := bm.restorePolicy
+	if r == nil {
+		if err = bm.s.Init(); err != nil {
+			return err
+		}
+	} else {
+		bm.logger.Infof("restoring cluster from existing cluster (%s)", r.BackupClusterName)
+		if err = bm.s.Clone(r.BackupClusterName); err != nil {
```

Comment:
this looks worse. why indeed do i have to call clone? this line itself is confusing.